pandas/_libs/tslibs/parsing.pyx

```diff
@@ -361,6 +371,33 @@ cdef parse_datetime_string_with_reso(
     if parsed is not None:
         return parsed, reso
 
+    # Try iso8601 first, as it handles nanoseconds
+    # TODO: does this render some/all of parse_delimited_date redundant?
+    string_to_dts_failed = _string_to_dts(
+        date_string, &dts, &out_bestunit, &out_local,
+        &out_tzoffset, False
+    )
+    if not string_to_dts_failed:
+        if dts.ps != 0 or out_local:
+            # TODO: the not-out_local case we could do without Timestamp;
+            #  avoid circular import
+            from pandas import Timestamp
+            parsed = Timestamp(date_string)
```

Comment:
So the `dts` components cannot be passed into `Timestamp` and `date_string` needs to be passed instead to go through the string parsing flow?

Reply:
i punted on trying to optimize the Timestamp construction (i.e. avoid re-parsing) bc i think this function needs a non-trivial refactor which is out of scope